service/rating.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require_relative 'file_handler'
+
+# Description/Explanation of Rating Class
+class Rating
+  attr_reader :orders, :books, :authors, :clients
+
+  def initialize(orders, books, authors, clients)
+    @orders = orders
+    @books = books
+    @authors = authors
+    @clients = clients
+  end
+
+  def top_ids(key_name, ntimes)
+    @orders.group_by { |hash| hash[key_name] }
+           .transform_values(&:count)
+           .sort_by { |_key, count| - count }[0...ntimes]
+           .to_h
+           .keys
+  end
+
+  def top_books(ntimes)
+    if @orders.instance_of?(Integer) || @orders.empty?
```

Comment:
Why you paste it here? It should be in FileHandler and it should work for all files

Reply:
We discussed about it before and resolved this question